pkg/image/unprivileged/resolver.go

```diff
@@ -0,0 +1,51 @@
+package unprivileged
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	auth "github.com/deislabs/oras/pkg/auth/docker"
+)
+
+func NewResolver(username, password string, configs ...string) (remotes.Resolver, error) {
+	// TODO(njhale): make plainHTTP and insecure optional
+	plainHTTP := true
+	opts := docker.ResolverOptions{
+		PlainHTTP: true,
+	}
+
+	client := http.DefaultClient
+	insecure := true
+	if insecure {
+		client.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		}
+	}
+	opts.Client = client
+
+	if username != "" || password != "" {
+		opts.Credentials = func(hostName string) (string, string, error) {
+			return username, password, nil
+		}
+		return docker.NewResolver(opts), nil
+	}
+
+	cli, err := auth.NewClient(configs...)
```

Comment:
does this support containerd's keychain integration (so you don't have to have your password in plaintext?)

Reply:
[looks like it does](https://github.com/deislabs/oras/blob/19a0d4482aed33ec4431dc8fa126a484c475f2db/pkg/auth/docker/client.go#L38)